.github/workflows/rust.yml

```
@@ -11,16 +12,114 @@ env:
+
+      - name: Cache target dir (per profile)
+        uses: actions/cache@v4
+        with:
+          path: target
+          key: ${{ runner.os }}-cargo-target-${{ matrix.profile }}-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-target-${{ matrix.profile }}-
+            ${{ runner.os }}-cargo-target-
+
+      - name: Setup Rust toolchain
+        uses: actions-rs/toolchain@v1
```

actionlint (1.7.12)

[error] the runner of "actions-rs/toolchain@v1" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)